group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1009,20 +1025,23 @@ public CompletableFuture<DescribeShareGroupOffsetsResponseData> describeShareGro
                     future.completeExceptionally(new IllegalStateException("Result is null for the read state summary"));
                     return;
                 }
-                List<DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseTopic> describeShareGroupOffsetsResponseTopicList =
-                    result.topicsData().stream().map(
-                        topicData -> new DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseTopic()
-                            .setTopicId(topicData.topicId())
-                            .setTopicName(metadataImage.topics().topicIdToNameView().get(topicData.topicId()))
-                            .setPartitions(topicData.partitions().stream().map(
-                                partitionData -> new DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponsePartition()
-                                    .setPartitionIndex(partitionData.partition())
-                                    .setStartOffset(partitionData.startOffset())
-                                    .setErrorMessage(partitionData.errorMessage())
-                                    .setErrorCode(partitionData.errorCode())
-                            ).toList())
-                    ).toList();
-                future.complete(new DescribeShareGroupOffsetsResponseData().setResponses(describeShareGroupOffsetsResponseTopicList));
+                result.topicsData().forEach(topicData ->
+                    describeShareGroupOffsetsResponseTopicList.add(new DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseTopic()
+                        .setTopicId(topicData.topicId())
+                        .setTopicName(requestTopicIdToNameMapping.get(topicData.topicId()))
+                        .setPartitions(topicData.partitions().stream().map(
+                            partitionData -> new DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponsePartition()
+                                .setPartitionIndex(partitionData.partition())
+                                .setStartOffset(partitionData.startOffset())
+                                .setErrorMessage(Errors.forCode(partitionData.errorCode()).message())
+                                .setErrorCode(partitionData.errorCode())
+                        ).toList())
+                    ));
+
+                future.complete(
+                    new DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseGroup()
+                        .setGroupId(requestData.groupId())
+                        .setTopics(describeShareGroupOffsetsResponseTopicList));
             });
```

Comment:
For my understanding: else where in the GroupCoorrdinatorService we schedule read operation over runtime but we handle this current code differently, why?

Reply:
This is something different. It's not reading from the `__consumer_offsets` topic in a way that is compatible with the timeline hash maps. What we need is a long-running operation that interacts with the share coordinator and returns a future. So, I think this is a viable way of achieving this end.

However, I will be making a further PR as part of https://issues.apache.org/jira/browse/KAFKA-18761 which *will* use the `GroupCoordinatorShard` and schedule a read operation in order to obtain the list of topic-partitions when they are not supplied in the request.

Reply:
Thanks for explaining.